Sources/ParseSwift/Parse Types/BaseParseUser.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  File 2.swift
+//  
+//
+//  Created by Pranjal Satija on 7/19/20.
+//
+
+import Foundation
+
+/// Used internally to form a concrete type representing `ParseUser`.
+/// `ParseUser` itself has associatedtype requirements, so it's often awkard to use it directly.
+struct BaseParseUser: ParseUser {
```

Comment:
Can you explain why this would be needed? Ir looked like the UserType that had the CurrentUserContainer and LoginInfo was handling this fine

Reply:
On second thought, it's not needed. I can use `AnyCodable` instead of doing tricky stuff with generics and this type. I'll make that change tomorrow!

Reply:
@cbaker6 so I've been trying to find a good solution to this, and I honestly think the `BaseParseUser` solution is the best one. Here's why:

* `AnyCodable` doesn't work unless the type you're encoding is one of a few predefined ones (see the implementation for `encode(to: Encoder)` in `AnyEncodable.swift`. I didn't know this when I said `AnyCodable` would solve the issue here.
* Without `AnyCodable`, `CurrentUserContainer` has to be generic because `ParseUser` has associated type and Self requirements, so we can't type something as `ParseUser` directly.
* If `CurrentUserContainer` is generic, then we need a concrete type within the SDK that we can use to instantiate a `CurrentUserContainer` and get the auth token. Outside of the SDK, it's not necessary, because SDK users will have their own concrete implementations of `ParseUser`.

All of these things put together make it pretty hard to store an arbitrary `ParseUser`. I think the `BaseParseUser` solution is good because when we're *setting* the current user, we can use the concrete type of `ParseUser` that's available when signing up, logging in, etc. When we're *getting* it, we can use `BaseParseUser` because anything that conforms to `ParseUser` should be decodable to a `BaseParseUser`.

Please let me know if you have any other ideas!

Reply:
Seems reasonable, I don't know how I feel about the `Base` part, but I currently can't think of something better.

I'm going to look back at my previous comment to remove it as I was looking for this exchange at the beginning of the review